### PR TITLE
UCT/DC: Increase DCI index to 16 bits

### PR DESCRIPTION
**Pull request by shasson5**
## What?
Increase DCI index to 16 bits

## Why?
Support large number of DCIs

### CODE REVIEW

**Review by yosefe — APPROVED** [review]

**Review by roiedanino — APPROVED** [review]

File: src/uct/ib/mlx5/dc/dc_mlx5.h [hunk 1]
```diff
@@ -64,6 +64,7 @@ struct ibv_ravh {
 
 typedef struct uct_dc_mlx5_ep     uct_dc_mlx5_ep_t;
 typedef struct uct_dc_mlx5_iface  uct_dc_mlx5_iface_t;
+typedef uint16_t                  uct_dci_index_t;
```
Comment (gleon99) [inline src/uct/ib/mlx5/dc/dc_mlx5.h:67]: `_dc_mlx5_`

Comment (roiedanino) [reply src/uct/ib/mlx5/dc/dc_mlx5.h:67]: I would prefer keeping it as short as possible, its being used in many places

Comment (shasson5) [reply src/uct/ib/mlx5/dc/dc_mlx5.h:67]: IMO would be too long and clumsy

**Comment by yosefe** [comment]
@shasson5 can you pls squash?

**Pull request merged**